### PR TITLE
Adding stat modifiers to the unit and calculations for getting a stat's actual value.

### LINKED ISSUE #9: Update models with stat modifiers

### LINKED ISSUE #10: Reset stat modifiers method at beginning of each round

### PR DESCRIPTION
**Pull request by Arcontos**
Went ahead and made the base stats non-final and added setters since we know some equipment/special rules will modify the base values.

Added modifiers for the unit's base stats but not the mount (I'm assuming nothing affects a mount's stats but if it does then I can do the same for the mounts).

Added methods to get the actual value of a stat (base stat + modifier).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -48,4 +48,10 @@ task generateMigration {
 		def migrationId = DateTimeFormatter.ofPattern('yyyyMMddkkmmssSSS').format(ZonedDateTime.now(ZoneOffset.UTC))
 		new File("src/main/resources/db/migration/V${migrationId}__${mName}.sql").write('')
 	}
+}
+
+flyway {
+	url = 'jdbc:postgresql://localhost:5432/T9AgeCombat'
```
Comment (Ggoering) [inline build.gradle]: this works for me too if 'postgres' is too generic

File: src/main/java/com/GG/T9AgeCombat/models/Unit.java [hunk 2]
```diff
@@ -188,4 +229,154 @@ public void updateExtraRank(Integer extraRank) {
...
+        defensiveWeaponSkillModifier = 0;
+        toughnessModifier = 0;
+        armorModifier = 0;
+        initiativeModifier = 0;
+        offensiveWeaponSkillModifier = 0;
+        attacksModifier = 0;
+        strengthModifier = 0;
+        armorPenetrationModifier = 0;
+    }
+
```
Comment (Ggoering) [inline src/main/java/com/GG/T9AgeCombat/models/Unit.java:247]: nice

File: src/main/java/com/GG/T9AgeCombat/predicates/DetermineModificationPredicate.java [hunk 3]
```diff
@@ -12,7 +12,7 @@ private DetermineModificationPredicate() {
 
     public static void applyBonus(Unit unit, String modification, int value) {
         if (modification.equals(STRENGTH.toString())) {
-            unit.updateStrength(value);
+            unit.updateStrengthModifier(value);
```
Comment (Ggoering) [inline src/main/java/com/GG/T9AgeCombat/predicates/DetermineModificationPredicate.java]: so I am not understanding why @nonfinal was given to all the stats when this is the pattern?  I didn't see anywhere that the base strength or other stat of a unit was being modified

**Review by Ggoering — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (Arcontos) [reply build.gradle]: I  can swap it back to what you had in your PR.

**Review by Ggoering — APPROVED** [review]

**Pull request merged**